### ISSUE TITLE
Deterministic PRNG in temperature sampling — every generation is identical

### ISSUE THREAD

**Issue by web3guru888** [web3guru888/ATLAS#14]
## Summary

`sample_from_probs()` uses the **step number** (0, 1, 2, ...) as the LCG seed, making every generation call with the same prompt produce **identical output**. There is no entropy source.

## Code

```rust
// crates/atlas-model/src/lib.rs:1068
sample_from_probs(&scaled, step as u64)

// line 1634
fn sample_from_probs(probs: &[f32], seed: u64) -> usize {
    let rand = lcg_uniform(seed);  // ← seed is just the step number
    // ...
}

// line 1645
fn lcg_uniform(seed: u64) -> f32 {
    let x = seed.wrapping_mul(6364136223846793005).wrapping_add(1442695040888963407);
    (x >> 33) as f32 / (1u64 << 31) as f32
}
```

`lcg_uniform` is a **one-shot hash**, not a stateful PRNG. Each step independently hashes its step number:
- Step 0 → `lcg_uniform(0)` → always 0.6710...
- Step 1 → `lcg_uniform(1)` → always 0.9653...
- Step 2 → `lcg_uniform(2)` → always 0.2596...
- etc.

### 1. No randomness between calls
Two API requests with the same prompt and temperature always produce the **exact same output**. Users expect different completions on repeated calls.

### 2. Predictable sampling pattern
The random values follow a fixed sequence that has nothing to do with the model's probability distribution. Step 0 always picks from the same quantile of the distribution regardless of context.

### 3. OpenAI API incompatibility
The OpenAI API spec says `temperature > 0` should produce varied output. Clients that retry on bad output get the same bad output.

### Option A: Stateful XorShift64 (minimal, zero-dep)
```rust
struct Rng(u64);
impl Rng {
    fn new() -> Self {
        // Seed from system time + pointer address for entropy
        let t = std::time::SystemTime::now()
            .duration_since(std::time::UNIX_EPOCH)
            .unwrap().as_nanos() as u64;
        Self(t ^ 0xdeadbeef)
    }
    fn next_f32(&mut self) -> f32 {
        self.0 ^= self.0 << 13;
        self.0 ^= self.0 >> 7;
        self.0 ^= self.0 << 17;
        (self.0 >> 33) as f32 / (1u64 << 31) as f32
    }
}
```

### Option B: Thread-local RNG
Store a `Rng` in `OlmoModel` (or thread-local), seed once at creation, advance each sample.

### generate() change
```rust
pub fn generate(&mut self, prompt: &[u32], max_new: usize, temperature: f32) -> Vec<u32> {
    // ...
    for step in 0..max_new {
        let next = if temperature < 1e-6 {
            argmax(&last_logits)
        } else {
            let mut scaled = ...;
            softmax_inplace(&mut scaled);
            sample_from_probs(&scaled, &mut self.rng)  // stateful RNG
        };
        // ...
    }
}
```

## Environment
- ATLAS v4.0.4
- Affects atlas-model and atlas-api
- All models, all temperatures > 0

**Comment by web3guru888** [comment]
Fixed in commit e831935.

**Changes:**
- Added stateful XorShift64 PRNG to atlas-model (replaces step-based LCG hash)
- Re-seeded from system time (nanoseconds) at each `generate()` call
- `sample_from_probs()` takes f32 random value from stateful RNG (not step-based hash)
- Repeated calls with same prompt + temp produce different output

**Tests:** `generate_temperature_varies`, `rng_xorshift_*`, `sample_from_probs_basic` ✅

**Note:** OLMo-3-7B produces repetitive output (near-100% mass on one token) — this is a model quality issue needing repetition penalty, not a PRNG bug.
